### ISSUE TITLE
Include non-document errors in command line checker

### ISSUE THREAD

**Issue by jaimeiniesta** [validator/validator#460]
When checking an URL that responds with a 404 error on the command line checker, it results in an uncaught java exception:

```
$ java -jar ./node_modules/vnu-jar/build/dist/vnu.jar --exit-zero-always --format json http://example.com/not
Exception in thread "main" nu.validator.xml.PrudentHttpEntityResolver$ResourceNotRetrievableException: http://example.com/not: HTTP resource not retrievable. The HTTP status from the remote server was: 404.
	at nu.validator.xml.PrudentHttpEntityResolver.resolveEntity(PrudentHttpEntityResolver.java:276)
	at nu.validator.validation.SimpleDocumentValidator.checkHttpURL(SimpleDocumentValidator.java:429)
	at nu.validator.client.SimpleCommandLineValidator.checkFiles(SimpleCommandLineValidator.java:248)
	at nu.validator.client.SimpleCommandLineValidator.main(SimpleCommandLineValidator.java:203)
```

Instead, it would be nice if a non-document error message was returned, as in the output from the web:

https://validator.nu/?doc=http://example.com/not&out=json

```json
{
  "url": "http://example.com/not",
  "messages": [
    {
      "type": "non-document-error",
      "subType": "io",
      "message": "HTTP resource not retrievable. The HTTP status from the remote server was: 404."
    }
  ]
}
```

**Comment by sideshowbarker** [comment]
Fixed and pushed as 17.1.32 to https://sideshowbarker.net/nightlies/ and to npm on the `dev` tag.

**Comment by jaimeiniesta** [comment]
Thank you, I tried it and it works fine with 404 responses.

There's another case, for unknown hosts the CLI outputs:

```json
{
    "messages": [
        {
            "type": "error",
            "url": "http://lkjlkjlkjkljljjkl.com",
            "message": "java.net.UnknownHostException: lkjlkjlkjkljljjkl.com: unknown error"
        }
    ]
}
```

While the web outputs a non-document-error, as it should:

https://checker.html5.org/?doc=http%3A%2F%2Fhiiiuhiiuhiuhiuhohhoih.com&out=json

```json
{
    "url": "http://hiiiuhiiuhiuhiuhohhoih.com",
    "messages": [
        {
            "type": "non-document-error",
            "subType": "io",
            "message": "hiiiuhiiuhiuhiuhohhoih.com: Name or service not known"
        }
    ]
}
```

Maybe for timeouts there's also a mismatch?

**Comment by sideshowbarker** [comment]
> There's another case, for unknown hosts the CLI outputs:

Fixed and pushed as 17.1.33 to https://sideshowbarker.net/nightlies/ and to npm on the dev tag.

> Maybe for timeouts there's also a mismatch?

I dunno. But if there’s a test case, I can check
